vowpalwabbit/loss_functions.cc

```diff
@@ -358,6 +358,58 @@ class poisson_loss : public loss_function
   }
 };
 
+class modified_hubber_loss : public loss_function
+{
+ public:
+  std::string getType() { return "modified_hubber"; }
+
+  float getLoss(shared_data*, float prediction, float label)
+  {
+    if (label != -1.f && label != 1.f)
+      std::cout << "You are using label " << label << " not -1 or 1 as loss function expects!" << std::endl;
+     float e = max(0, 1-prediction*label);
+    return (prediction*label >= -1.f) ? e*e : -4*label*prediction;
+  }
+
+  float getUpdate(float prediction, float label, float update_scale, float pred_per_update)
+  {
+
+  }
+
```

Comment:
I don't understand how updates work so anyone can explain then I will update it accordingly.